components/automate-deployment/pkg/services/services.go

```diff
@@ -102,13 +102,6 @@ func ValidateProductDeployment(products []string) error {
 		}
 	}
 
-	for requiredCollectionName, requiredCollection := range requiredCollectionSet {
```

Comment:
This prevented dependencies of type "product" from the products.meta file from automatically being install. Before the command to install desktop would have been `chef-automate deploy --product automate --product desktop` with this removed it is `chef-automate deploy --product desktop`. 
Desktop is defined below
```
    {
      "name": "desktop",
      "type": "product",
      "dependencies": ["automate"],
      "services": []
    }
```
With this code removed the dependence "automate" is automatically installed without explicitly declaring to install it.